megamek/src/megamek/common/LosEffects.java

```diff
@@ -1174,38 +1192,42 @@ private static LosEffects losForCoords(Game game, AttackInfo ai,
                 //number of screens doesn't matter. One is enough to block
                 los.screen++;
             }
-            //heavy industrial zones can vary in height up to 10 levels, so lets
-            //put all of this into a for loop
-            int industrialLevel = hex.terrainLevel(Terrains.INDUSTRIAL);
-            if (industrialLevel != Terrain.LEVEL_NONE) {
-                for (int level = 1; level < 11; level++) {
-                    if ((hexEl + level > maxUnitHeight)
-                            || ((hexEl + level > ai.attackAbsHeight) && attackerAdjc)
-                            || ((hexEl + level > ai.targetAbsHeight) && targetAdjc)) {
-                        // check industrial zone
-                        if (industrialLevel == level) {
-                            los.heavyIndustrial++;
+
+            if (!ai.lowAltitude) {
+                //heavy industrial zones can vary in height up to 10 levels, so lets
+                //put all of this into a for loop
+                int industrialLevel = hex.terrainLevel(Terrains.INDUSTRIAL);
+                if (industrialLevel != Terrain.LEVEL_NONE) {
+                    for (int level = 1; level < 11; level++) {
+                        if ((hexEl + level > maxUnitHeight)
+                                || ((hexEl + level > ai.attackAbsHeight) && attackerAdjc)
+                                || ((hexEl + level > ai.targetAbsHeight) && targetAdjc)) {
+                            // check industrial zone
+                            if (industrialLevel == level) {
+                                los.heavyIndustrial++;
+                            }
                         }
                     }
                 }
-            }
-            //planted fields only rise one level above the terrain
-            if (hex.containsTerrain(Terrains.FIELDS)) {
-                if (((hexEl + 1 > ai.attackAbsHeight) && (hexEl + 2 > ai.targetAbsHeight))
-                        || ((hexEl + 1 > ai.attackAbsHeight) && attackerAdjc)
-                        || ((hexEl + 1 > ai.targetAbsHeight) && targetAdjc)) {
-                    los.plantedFields++;
+                //planted fields only rise one level above the terrain
+                if (hex.containsTerrain(Terrains.FIELDS)) {
+                    if (((hexEl + 1 > ai.attackAbsHeight) && (hexEl + 2 > ai.targetAbsHeight))
+                            || ((hexEl + 1 > ai.attackAbsHeight) && attackerAdjc)
+                            || ((hexEl + 1 > ai.targetAbsHeight) && targetAdjc)) {
+                        los.plantedFields++;
 
+                    }
                 }
             }
-            
+
+
             // Intervening Smoke and Woods
-            int smokeLevel = hex.terrainLevel(Terrains.SMOKE);
             int woodsLevel = hex.terrainLevel(Terrains.WOODS);
             int jungleLevel = hex.terrainLevel(Terrains.JUNGLE);
             int foliageElev = hex.terrainLevel(Terrains.FOLIAGE_ELEV);
```

Comment:
maybe, Terrains.getTerrainElevation() can be used to find foliageElev instead of the terrain level with lowAltitude as parameter so the entire Board change is unnecessary?

Reply:
I'll check it out.

Reply:
Incorporated `getTerrainElevation()` where possible.